kratos/input_output/stl_io.cpp

```diff
@@ -23,47 +23,74 @@
 namespace Kratos
 {
 
-StlIO::StlIO(std::filesystem::path const& Filename, const Flags Options)
-    : mOptions(Options)
+StlIO::StlIO(std::filesystem::path const& Filename, Parameters ThisParameters):
+    mParameters(ThisParameters)
 {
-    Kratos::shared_ptr<std::fstream> pFile = Kratos::make_shared<std::fstream>();
+    mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+    Kratos::shared_ptr<std::fstream> p_file = Kratos::make_shared<std::fstream>();
 
     // set default mode to read
-    std::fstream::openmode OpenMode;
+    std::fstream::openmode open_mode;
 
+    const std::string open_mode_str = mParameters["open_mode"].GetString();
     // handle other mode options
-    if (mOptions.Is(IO::READ)) {
-        OpenMode = std::fstream::in;
-    } else if (mOptions.Is(IO::APPEND)) {
-        OpenMode = std::fstream::in | std::fstream::app;
-    } else if (mOptions.Is(IO::WRITE)) {
-        OpenMode = std::fstream::out;
+    if (open_mode_str == "read") {
```

Comment:
Why not flag?

Reply:
Legacy, I fully agree with parameters

Reply:
It is fully agreed with @KratosMultiphysics/technical-committee ?, I will remove my block